touchnet/models/kimi_audio/modeling_kimi_audio.py

```
@@ -0,0 +1,1212 @@
+        self.embed_positions2 = nn.Embedding(max_source_positions, self.config.d_model)
+        self.embed_positions2.weight.data.copy_(self.embed_positions.weight.data[:max_source_positions])
+        # NOTE(xcsong): we do not need ema for inference, keep it for disable the warning of missing weight
+        if config.quantize_ema_decay is not None:
+            self.codebook.weight.requires_grad = False
+            self.register_buffer("ema_count", torch.ones(config.quantize_vocab_size, dtype=torch.float))
+            self.register_buffer("ema_weight", self.codebook.weight.data.clone().float())
+
+    def _freeze_parameters(self):
+        for param in self.parameters():
+            param.requires_grad = False
+        self._requires_grad = False
```

Pylint (3.3.7)

[warning] Attribute '_requires_grad' defined outside __init__

(W0201)